c++/src/kj/refcount.h

```diff
@@ -230,6 +232,76 @@ class Rc {
   friend class EnableAddRefToThis;
 };
 
+template <typename T>
+class Rc<const T> {
+  // Variation on Rc that enables copy-on-write semantics. The held pointer
+  // is const. When copy() is called, the held T is copied to a mutable instance.
+public:
+  KJ_DISALLOW_COPY(Rc);
+  Rc() { }
+  Rc(decltype(nullptr)) { }
+  inline Rc(Rc&& other) noexcept = default;
+
+  kj::Own<const T> toOwn() {
+    // Convert Rc<const T> to Own<const T>.
+    // Nullifies the original Rc<const T>.
+    return kj::mv(own);
+  }
+
+  Rc<T> copy() {
+    return refcounted<T>(*own);
```

Comment:
This, of course, assumes `T` implements a copy constructor, which `Refcounted` by default implicitly deletes. Should we look for a `clone()` member method instead? Or have the option to pass in a static copy function as part of the template?